torch/csrc/Exceptions.h

```diff
@@ -7,6 +7,7 @@
 #include <mutex>
 
 #include <c10/util/Exception.h>
+#include <c10/cuda/CUDACachingAllocator.h>
```

Comment:
It looks like this include isn't allowed. Probably the easiest way to fix this is to move the declaration of CUDAOutOfMemoryError into c10/util/Exception.h so it's no longer in c10/cuda, keep everything else as sis.